pjsip/src/pjsua-lib/pjsua_media.c

```diff
@@ -4384,6 +4390,16 @@ pj_status_t pjsua_media_channel_update(pjsua_call_id call_id,
     pj_memcpy(call->media, call->media_prov,
               sizeof(call->media_prov[0]) * call->med_prov_cnt);
 
+    /* Create synchronizer */
+    if ((maudcnt+mvidcnt) > 1 && !call->av_sync) {
```

Comment:
Do we need pjsua config to enable/disable the sync, as well as to choose which media will be sync?

Reply:
Yes, enable/disable should be needed. As this is a new feature, I guess we can postpone fine-tuning settings such as selecting which media to sync, allowing two or more synchronizers in a call, based on user feedback?

Reply:
The thing is, I'm not sure if text stream should be synchronized by default, so that's why I proposed the setting to let the user decide.

Having said that, I'm not sure whether we should also implement text sync, so I suppose enable/disable is sufficient for now, with doc noting that it currently only applies for audio and video.

Reply:
In the future, text sync may be useful for streaming a video with subtitle (may require something like text-player, similar to wav/video-player). However, for now I think having text sync is not so urgent.

Just to add a sample scenario of two or more synchronizers in a call, in a conference call, there can be two sets of audio+video stream (so totally there will be 4 streams), rather than synchronize all 4 streams using one synchronizer, perhaps it is better to employ a separate synchronizer for each set (due to different original sources, they may have different codec delay, network delay, NTP source, etc). So in the future, when designing fine tune settings per media, perhaps we need to also consider this scenario.